js/deribit.js

```diff
@@ -458,28 +458,66 @@ module.exports = class deribit extends Exchange {
                 const base = this.safeCurrencyCode (baseId);
                 const quote = this.safeCurrencyCode (quoteId);
+                const settle = this.safeCurrencyCode (settleId);
                 const type = this.safeString (market, 'kind');
-                const future = (type === 'future');
+                const settlementPeriod = this.safeValue (market, 'settlement_period');
+                const swap = (settlementPeriod === 'perpetual');
+                const future = !swap && (type === 'future');
                 const option = (type === 'option');
-                const active = this.safeValue (market, 'is_active');
+                let symbol = quote + '/' + base + ':' + settle;
+                let expiry = undefined;
+                let strike = undefined;
+                let optionType = undefined;
+                if (option || future) {
+                    expiry = this.yyyymmdd (this.iso8601 (this.safeInteger (market, 'expiration_timestamp')), '');
+                    symbol = symbol + '-' + expiry;
+                    if (option) {
+                        strike = this.safeNumber (market, 'strike');
+                        optionType = this.safeString (market, 'option_type');
+                        symbol = symbol + ':' + strike + ':' + optionType;
+                    }
+                }
                 const minTradeAmount = this.safeNumber (market, 'min_trade_amount');
                 const tickSize = this.safeNumber (market, 'tick_size');
-                const precision = {
-                    'amount': minTradeAmount,
-                    'price': tickSize,
-                };
                 result.push ({
                     'id': id,
-                    'symbol': id,
+                    'symbol': symbol,
                     'base': base,
                     'quote': quote,
-                    'active': active,
-                    'precision': precision,
+                    'settle': settle,
+                    'baseId': baseId,
+                    'quoteId': quoteId,
+                    'settleId': settleId,
+                    'type': type,
+                    'spot': false,
+                    'margin': false,
+                    'swap': true,
+                    'future': future,
+                    'option': option,
+                    'derivative': true,
+                    'contract': true,
+                    'linear': false,
+                    'inverse': true,
                     'taker': this.safeNumber (market, 'taker_commission'),
                     'maker': this.safeNumber (market, 'maker_commission'),
+                    'contractSize': this.safeString (market, 'contract_size'),
+                    'active': this.safeValue (market, 'is_active'),
+                    'expiry': expiry,
+                    'strike': strike,
+                    'optionType': optionType,
+                    'fees': this.safeValue (this.fees, 'trading'),
```

Comment:
This is empty, I'm not sure what the proper way to set it is

Reply:
If we have per-market fees, we could fill them in here, if not, we could just remove this field cause it will be populated from `this.fees` if those are set.

Reply:
I'd like to keep it in there for the sake of keeping things in the same order across exchanges. I'm trying to make the order consistent as I unify all the fetchMarkets methods

Reply:
This line is not necessary, cause it's done here: https://github.com/ccxt/ccxt/blob/master/js/base/Exchange.js#L692